src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/resources/viewmodel/ResourcesViewModel.kt

```diff
@@ -1,43 +1,48 @@
 package org.wycliffeassociates.otter.jvm.app.ui.resources.viewmodel
 
 import javafx.application.Platform
-import javafx.beans.property.SimpleObjectProperty
-import javafx.beans.property.SimpleStringProperty
 import org.wycliffeassociates.otter.common.data.workbook.*
+import org.wycliffeassociates.otter.common.domain.content.Recordable
 import org.wycliffeassociates.otter.common.utils.mapNotNull
+import org.wycliffeassociates.otter.jvm.app.ui.resourcetakes.viewmodel.TakesViewModel
+import org.wycliffeassociates.otter.jvm.app.ui.workbook.viewmodel.WorkbookViewModel
 import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.ResourceGroupCardItemList
 import org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model.resourceGroupCardItem
 import tornadofx.*
 
 class ResourcesViewModel : ViewModel() {
-    val activeWorkbookProperty = SimpleObjectProperty<Workbook>()
-    val workbook: Workbook
-        get() = activeWorkbookProperty.value
-
-    val activeChapterProperty = SimpleObjectProperty<Chapter>()
-    val chapter: Chapter
-        get() = activeChapterProperty.value
-
-    val activeResourceSlugProperty = SimpleStringProperty()
-    val resourceSlug: String
-        get() = activeResourceSlugProperty.value
+    private val takesViewModel: TakesViewModel by inject()
+    private val workbookViewModel: WorkbookViewModel by inject()
 
     val resourceGroups: ResourceGroupCardItemList = ResourceGroupCardItemList(mutableListOf())
```

Comment:
If this is a usual use case, perhaps the initialization of an empty mutable list could be done by default and thus not have a parameter here?

Reply:
Sure, good point